File: velour.go
package main

import (
	"fmt"

	"velour/seqio"
	"velour/test/time"

	// "github.com/pkg/profile"
)

func main() {
	// defer profile.Start().Stop()

	fragments := []string{"data/staphylococcus_aureus/frag1.fastq", "data/staphylococcus_aureus/frag2.fastq"}

	k := seqio.EstimateK(2900000)
	fmt.Println("Estimated K:", k)

	test_time.TimeAll(fragments[:1], k)
}

File: src/Velour.jl
module Velour

	include("KmerIO.jl")
	using .KmerIO
	using ProfileView

	function main()
		fragments = ["data/staphylococcus_aureus/frag1.fastq", "data/staphylococcus_aureus/frag2.fastq"]
		# graph = KmerIO.DeBruijnGraph.new_graph()

		for fragment in fragments
			KmerIO.graph_from_fastq(fragment)
			# println(graph.num_nodes)
			break
		end
	end

end
